training/tf/parse.py

```diff
@@ -270,21 +274,36 @@ def main(args):
     if not chunks:
         return
 
-    parser = ChunkParser(chunks)
+    # The following assumes positions from one game are not
+    # spread through chunks.
+    random.shuffle(chunks)
+    training, test = split_chunks(chunks, 0.1)
+    print("Training with {0} chunks, validating on {1} chunks".format(
+        len(training), len(test)))
 
-    run_test(parser)
+    #run_test(parser)
     #benchmark(parser)
 
+    train_parser = ChunkParser(training)
     dataset = tf.data.Dataset.from_generator(
-        parser.parse_chunk, output_types=(tf.string))
+        train_parser.parse_chunk, output_types=(tf.string))
     dataset = dataset.shuffle(1 << 18)
     dataset = dataset.map(_parse_function)
     dataset = dataset.batch(BATCH_SIZE)
     dataset = dataset.prefetch(4)
     iterator = dataset.make_one_shot_iterator()
-    next_batch = iterator.get_next()
+    next_train_batch = iterator.get_next()
+
+    test_parser = ChunkParser(test, 1)
+    dataset = tf.data.Dataset.from_generator(
+        test_parser.parse_chunk, output_types=(tf.string))
```

Comment:
This does need a shuffle buffer too. Without shuffling, testing 800 batches of 512 positions means we test about 1024 different games, which means the errors on the MSE measurement are very correlated.

Reply:
Actually, a conditional skip would be much handier here than a shuffle buffer to fix that, and more memory friendly.